Assets/__Scripts/UI/LogLineUI.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using SimpleJSON;
+using TMPro;
+using UnityEngine;
+using UnityEngine.Networking;
+using UnityEngine.UI;
+using Debug = UnityEngine.Debug;
+
+public class LogLineUI : MonoBehaviour
+{
+    private const string DevKey = "zx634zuwuNuzKCocnPpTY99c2uhDJxr3";
+    private const string PasteeeKey = "uYsOA14Wo2JBxzQBgMHNfWOi6Mlbghchc8B86IwG6";
+    private const int LastLinesCount = 20;
+    
+    [SerializeField] internal TextMeshProUGUI TextMesh;
+    [SerializeField] internal Button ReportButton;
+    private string _previousMessages = "";
+    private DevConsole.Logline _logline;
+    private bool _sentReport;
+    
+    private static readonly string Seperator = new String('-', 50);
+
+    internal void SetupReport(DevConsole.Logline logline, List<string> lines)
+    {
+        _logline = logline;
+        ReportButton.gameObject.SetActive(logline.Type == LogType.Exception);
+        ReportButton.image.color = Color.cyan;
+        _previousMessages = logline.Type == LogType.Exception ? string.Join("\n", lines.Skip(lines.Count - LastLinesCount)) : "";
+        _sentReport = false;
+    }
+
+    public void SendReport()
+    {
+        if (!_sentReport)
+        {
+            _sentReport = true;
+            StartCoroutine(SendToPastebin());
+        }
+    }
+    
+    private static string GenerateSystemInfo()
+    {
+        return "APP: ChroMapper " + Application.version + ", Unity " + Application.unityVersion + " (" + Environment.CommandLine + ")\n" +
+               "CPU: " + SystemInfo.processorType + " (" + SystemInfo.processorCount + " cores)\n" +
+               "GPU: " + SystemInfo.graphicsDeviceName + "\n" +
+               "RAM: " + SystemInfo.systemMemorySize + " MB\n" +
+               "OS: " + SystemInfo.operatingSystem;
+    }
+
+    private string Heading(string text, bool first = false)
+    {
+        return (first ? "" : "\n\n\n") + $"{Seperator}\n{text}\n{Seperator}\n";
+    }
+
+    public IEnumerator SendToPastebin()
```

Comment:
Honestly I'd rename this to something like `GenerateBugReport`, especially when Pastebin isn't being used at the moment.

Reply:
This may be a derp moment